### PR TITLE
[EventHubs&ServiceBus] merge sb and eh pyamqp

### PR DESCRIPTION
**Pull request by swathipil**
* Merged _pyamqp files from ServiceBus and EventHubs, resolving all merge conflicts
fixes: #25936
fixes: #25471

TODO:
- [ ] check that mypy/pylint pass on CI for SB
- [ ] mypy/pylint for EH

### CODE REVIEW

File: sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/cbs.py [hunk 1]
```diff
@@ -169,7 +204,7 @@ def _cbs_link_ready(self):
         if self.state != CbsState.OPEN:
             return False
         if self.state in (CbsState.CLOSED, CbsState.ERROR):
-            raise TokenAuthFailure(
+            raise TokenAuthFailure( # pylint: disable = no-value-for-parameter
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/cbs.py:207]: @kashifkhan - this is expecting `status_code` and `status_description` as parameters. Should these be passed in?

File: sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_cbs_async.py [hunk 2]
```diff
@@ -158,7 +160,7 @@ async def _cbs_link_ready(self):
         if self.state != CbsState.OPEN:
             return False
         if self.state in (CbsState.CLOSED, CbsState.ERROR):
-            raise TokenAuthFailure(
+            raise TokenAuthFailure( # pylint: disable = no-value-for-parameter
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_cbs_async.py:163]: @kashifkhan - this is expecting `status_code` and `status_description` as parameters. Should these be passed in?

File: sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py [hunk 3]
```diff
@@ -420,26 +420,35 @@ async def negotiate(self):
...
             )
 
 
 class WebSocketTransportAsync(AsyncTransportMixin):
-    def __init__(self, host, port=WEBSOCKET_PORT, connect_timeout=None, ssl=None, **kwargs):
+    def __init__(
+        self,
+        host,
+        *,
+        port=WEBSOCKET_PORT,  # pylint: disable=unused-argument
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py:436]: @kashifkhan - looks like port isn't being used at all. checking that it's intentional?

File: sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_cbs_async.py (hunk 2 above)
Comment (kashifkhan) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_cbs_async.py:163]: `condition` should become `status_code` and `description` => 'status_description`

File: sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/cbs.py (hunk 1 above)
Comment (kashifkhan) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/cbs.py:207]: `condition` should become `status_code` and `description` => 'status_description`

File: sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py (hunk 3 above)
Comment (kashifkhan) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py:436]: It was never being used from the get go, I think we should append to the end of the host 
`self.host = f"{self.host}:{port}"`

File: sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_connection.py [hunk 4]
```diff
@@ -103,72 +103,43 @@ def __init__(self, endpoint, **kwargs):  # pylint:disable=too-many-statements
         if custom_endpoint_address:
             custom_parsed_url = urlparse(custom_endpoint_address)
             custom_port = custom_parsed_url.port or WEBSOCKET_PORT
-            custom_endpoint = "{}:{}{}".format(
-                custom_parsed_url.hostname, custom_port, custom_parsed_url.path
-            )
+            custom_endpoint = "{}:{}{}".format(custom_parsed_url.hostname, custom_port, custom_parsed_url.path)
```
Comment (kashifkhan) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_connection.py:106]: minor nit: f-String :)

File: sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_session_async.py [hunk 5]
```diff
@@ -90,10 +88,17 @@ async def _set_state(self, new_state):
...
+        _LOGGER.info(
+            "Session state changed: %r -> %r",
+            previous_state,
+            new_state,
+            extra=self.network_trace_params,
+        )
         futures = []
         for link in self.links.values():
-            futures.append(asyncio.ensure_future(link._on_session_state_change())) # pylint: disable=protected-access
+            futures.append(
```
Comment (kashifkhan) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_session_async.py]: same as below, we can make this in to list comprehension :)

Comment (swathipil) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_session_async.py]: will probably hold off on this until a separate PR where we fix list comprehension/fstrings/etc. ,like you mentioned.

Comment (kashifkhan) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_session_async.py]: agree

**Review by kashifkhan — APPROVED** [review]

**Pull request merged**